src/controller/CHIPDeviceController.cpp

```diff
@@ -3051,6 +3067,28 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
         }
     }
     break;
+    case CommissioningStage::kConfigureTCAcknowledgments: {
```

Comment:
This does not seem to quite match the spec.  The spec says that at this point, if the relevant feature is supported by server (which is not being checked), the commissioner SHALL present the T&Cs and whatnot.  That's not what this code is doing....

What seems expected here is that the UI is presented to the user before we ever get into this code.  It's not clear to me that this is technically spec-compliant, though it might not be obviously observable?

In either case, we should probably make the spec more lenient here, since we are discovering we cannot actually implement it as written.

Reply:
I updated this PR to focus on the cluster server (commissionee) implementation. I have a followup that will implement the cluster client (commissioner). In the previous revision of the PR, yes, the chip-tool required knowledge of the terms that are required, which could be accomplished if the VID/PID are known based on the scan information. In a followup to that, I am working on the Android CHIPTool, and just started to add the app layer callback to present the terms when the VID/PID are not known until announcement scan from the AutoCommissioner.

https://github.com/project-chip/connectedhomeip/compare/master...swan-amazon:connectedhomeip:feature/enhanced-setup-flow-android-chip-tool